streamz/collection.py

```diff
@@ -204,11 +206,13 @@ def accumulate_partitions(self, func, *args, **kwargs):
         if returns_state:
             _, example = example
         stream = self.stream.accumulate(func, *args, start=start,
-                returns_state=returns_state, **kwargs)
+                                        returns_state=returns_state, **kwargs)
 
-        for typ, s_type in _stream_types[stream_type]:
-            if isinstance(example, typ):
-                return s_type(stream, example)
+        try:
+            s_type = get_stream_type(example, stream_type)
```

Comment:
Having `get_stream_type` return `None` where the class is not found, and no try/except here seemed simple enough, why the change?